src/sage/algebras/quatalg/quaternion_algebra.py

```diff
@@ -1058,24 +1056,47 @@ def discriminant(self):
             except NotImplementedError:
                 raise ValueError("base field must be rational numbers or number field")
         else:
-            return hilbert_conductor(self._a, self._b)
+            return prod(self.ramified_primes())
 
-    def ramified_primes(self):
+    def ramified_primes(self, sorted=False):
```

Comment:
Also, I suppose this should be cached; simply add `@cached_method` above.

Reply:
One more comment: is there any use for `sorted=False` at all, apart from perhaps a tiny speedup?

Reply:
Good point! Regarding sorted: It's really only there for speeding up a bit, since e.g. the computation of the discriminant does not need the ordering and I believe that the comparison in `.is_isomorphic` should also be faster by comparing the lists as sets than when sorting them first.

Reply:
> Good point! Regarding sorted: It's really only there for speeding up a bit, since e.g. the computation of the discriminant does not need the ordering and I believe that the comparison in `.is_isomorphic` should also be faster by comparing the lists as sets than by sorting them first.

Correction to the above, I remembered that I also wanted to put it in there to easily mimic the behavior of the old variant, which correctly sorted the prime during factorization.

Reply:
If so, I recommend removing the option. It makes the output more clear to the user, and the time it takes to sort (I assume) < 100 elements is negligible anyways, compared to say the hilbert symbols.

Reply:
That's a fair point, I'll remove it (and adapt the isomorphism check accordingly)